File: src/asciinema.py
import time
import sys
import subprocess

STEP = 0.1
BIO = """👋 Bio
 • Gnu Believer. Support it now.
 • Tux Cultist(We suck Windows).
 • Disciple of RTFM & STFW. Ps. ATFL(Ask The Fucking LLM)
"""

CONTACT = """✉️ Contact me
 • Gmail - ■■■■■■■
   Please Sign the email with GPG.
 • Gvoice - ■■■■■■■
   Feel free to leave voice messages.
 • Telegram - ■■■■■■■
"""


def __type(content: str) -> None:
    for char in content:
        sys.stdout.write(char)
        sys.stdout.flush()
        time.sleep(STEP)


if __name__ == '__main__':
    __type(BIO)
    __type(CONTACT)

File: src/asciinema.js
const STEP = 100; // 100ms, equivalent to 0.1 seconds
const BIO = `👋 Bio
 • Gnu Believer. Support it now.
 • Tux Cultist(We suck Windows).
 • Disciple of RTFM & STFW. Ps. ATFL(Ask The Fucking LLM)
`;

const CONTACT = `✉️ Contact me
 • Gmail - ■■■■■■■
   Please Sign the email with GPG.
 • Gvoice - ■■■■■■■
   Feel free to leave voice messages.
 • Telegram - ■■■■■■■
`;

function type(content) {
    return new Promise((resolve) => {
        let i = 0;
        function typeChar() {
            if (i < content.length) {
                process.stdout.write(content[i]);
                i++;
                setTimeout(typeChar, STEP);
            } else {
                resolve();
            }
        }
        typeChar();
    });
}

async function main() {
    await type(BIO);
    await type(CONTACT);
}

main();